src/server/services/settings-services.ts

```diff
@@ -1,34 +1,38 @@
 import { Strapi } from '@strapi/strapi'
+import { CustomSettings } from '../../types'
 import { isValidApiKey } from '../utils/config'
 
 export default ({ strapi }: { strapi: Strapi }) => ({
-    async getConfig() {
+    async getSettings() {
         const pluginStore = strapi.store({
             environment: strapi.config.environment,
             type: 'plugin',
             name: 'api-video-uploader',
         })
 
-        const configKey = await pluginStore.get({
-            key: 'apiKey',
+        const customSettings = await pluginStore.get({
+            key: 'settings',
         })
```

Comment:
I'm afraid it's not backwards compatible. For users of the current version, the api key they are currently using will be lost, right? Maybe by using 2 keys (the old "apiKey" + a new one for the default value of private) we wouldn't have this problem ?

Reply:
This is true API keys will be lost. I prefer the separation - although there is then a question of how to handle the saving of the settings page if one setting succeeds and the other fails.